solcore/tests/test_absorption.py

```diff
@@ -285,3 +295,133 @@ def test_sopra_absorption():
+
+    A_nosubs = my_structure[0].layer_absorption
+
+
+    A = np.vstack((A_subs, A_nosubs))
+
+    A_data = np.array([[0.5209978, 0.62262739, 0.41829377],
+                       [0.5209978, 0.62284555, 0.37789382]])
+
+    assert all([d == approx(o) for d, o in zip(A, A_data)])
+
+def test_substrate_presence_profile():
+    wavelength = np.linspace(300, 800, 3) * 1e-9
+
+    GaAs = material("GaAs")(T=300)
+
+    my_structure = SolarCell([Layer(si(700, "nm"), material=GaAs)], substrate=GaAs)
+
+    solar_cell_solver(my_structure, 'optics',
+                      user_options={'wavelength': wavelength, 'optics_method': 'TMM',
+                                    'no_back_reflexion': False})
+
+    z_pos = np.linspace(0, my_structure.width, 10)
+
+    profile_subs = my_structure[0].absorbed(z_pos)
+
+    my_structure = SolarCell([Layer(si(700, "nm"), material=GaAs)])
+
+    solar_cell_solver(my_structure, 'optics',
+                      user_options={'wavelength': wavelength, 'optics_method': 'TMM',
+                                    'no_back_reflexion': False, })
+
+    profile_nosubs = my_structure[0].absorbed(z_pos)
+
+    profile = np.vstack((profile_subs, profile_nosubs))
+
+    profile_data = np.array([[4.35364278e+07, 4.29043895e+06, 9.37889190e+05],
+                             [6.28046841e+04, 2.51407523e+06, 8.40685958e+05],
+                             [9.05652587e+01, 1.47317317e+06, 7.53556838e+05],
+                             [1.30543325e-01, 8.63233564e+05, 6.75457758e+05],
+                             [1.88089178e-04, 5.05826779e+05, 6.05452855e+05],
+                             [3.22944240e-07, 3.00491300e+05, 5.44224778e+05],
+                             [4.65782106e-10, 1.76079153e+05, 4.87821069e+05],
+                             [6.71529364e-13, 1.03177016e+05, 4.37263024e+05],
+                             [9.67758605e-16, 6.04584451e+04, 3.91944804e+05],
+                             [0.00000000e+00, 0.00000000e+00, 0.00000000e+00],
+                             [4.35364278e+07, 4.25691019e+06, 4.84783054e+05],
+                             [6.28046841e+04, 2.52741110e+06, 6.48506623e+05],
+                             [9.05652587e+01, 1.51218752e+06, 9.69894260e+05],
+                             [1.30543325e-01, 8.88603973e+05, 5.31398525e+05],
+                             [1.88089178e-04, 4.92629409e+05, 2.35132516e+05],
+                             [3.22944240e-07, 2.62671365e+05, 6.95243874e+05],
+                             [4.65782107e-10, 1.39173748e+05, 6.70278004e+05],
+                             [6.71530801e-13, 1.01785699e+05, 1.45363256e+05],
+                             [9.69780702e-16, 1.00921869e+05, 2.87066651e+05],
+                             [0.00000000e+00, 0.00000000e+00, 0.00000000e+00]]
+                            )
+
+
+
+    assert all([d == approx(o) for d, o in zip(profile, profile_data)])
+
+
+# def test_custom_materials():
```

Comment:
Why is this tests commented out? If it is not longer relevant, simply deleted. Otherwise, if it depends on some work in progress that should be solved in the future but is not working yet, it is best to use:

```python
from pytest import mark
```

and then just before the test:

```python
@mark.skip
def test_custom_materials()
```

This way, the test is identified as such, but it is skipped, not filling the code with commented, un-used lines.

Reply:
We discussed this on Slack, it was because the test gave an error because it depends on downloading the database. I don't know how to resolve this. I guess I will just remove the test for now since this is quite a fundamental issue.

Reply:
OK, I remember now. I would leave it with the skip label as above and, maybe, add a TODO text explaining the problem:

```python
@mark.skip
def test_custom_materials()
# bla bla bla

# TODO: This is not working yet because of...
# bla bla bla. The code
```

Reply:
I would also open an issue, in case anyone can figure out how to do it.